src/util.js

```diff
@@ -0,0 +1,40 @@
+const tagSeparator = '-tags=';
+
+function isEmpty(value) {
+  return (
+    (typeof value == 'string' && !value.trim()) ||
+    typeof value == 'undefined' ||
+    value === null
+  );
+}
+
+function encodeKey(metricName, tags) {
+  if (this.isEmpty(tags)) {
+    return metricName;
+  }
+  let tagsType = Object.prototype.toString.call(tags).slice(8, -1);
+  if (tagsType === 'Object') {
```

Comment:
To check if a variable type is "object", see https://stackoverflow.com/a/8511350/1885433

Reply:
thanks for pointing this out, but `null` has already been checked in `isNil(tags)` thus cannot be null if it enters this line

Reply:
Yeah I agree with you on that but we can still do:
```suggestion
  if (typeof tags === "object") {
```

Reply:
oh...sorry for missing your point. I'll change the code accordingly.